repaygithub/cactus
# CACTUS-703 :: Table & DataGrid Margins

Ticket: https://repayonline.atlassian.net/browse/CACTUS-703

### Notes
- The Table was setting `min-width: calc(100% - 32px);` for `fullWidth`, but obviously that won't work for customizable margins (that 32px was to account for the 16px of margin on each side). I don't know if there's a reason it needs to be `min-width`, but I changed to `width: 100%;` here because for some reason `min-width: 100%;` doesn't seem to take margins into account. I don't think that's a problem but if anyone has more insight on why that was setting min width let me know.
- I'm certain that there will be updates to the DataGrid storyshots (I removed the style overrides from the DataGrid basic usage story) but I've been having issues building/running the storyshot tests. I started a [slack thread](https://repay.slack.com/archives/CABGTU4S3/p1635455104077300) about it but as of now I don't know how to resolve it. I wanted to get this pushed up so we could start getting eyes on it while I'm out on PTO though.

### Testing
1. Run the storybooks and bring up the `Table` basic usage story. Play around with the margin prop and inspect the table wrapper to make sure the margins are correctly being applied.
2. Open the `DataGrid` storybook and set the margin using the control. Inspect the DataGrid and make sure the margins are being applied. Set `fullWidth` to true and make sure everything looks good.
